### PR TITLE
Just a small update.

### PR DESCRIPTION
**Pull request by Loki214**
Added some missing function.
Removed some duplicates.
Removed Depreciated functions.
Fixed some obvious incorrect function descriptions.

### CODE REVIEW

File: src/defs/server.ts [hunk 1]
```diff
@@ -4,6 +4,51 @@ import {LuaFunction, MTAFunction, ScriptSide} from "./defs";
...
+tmpDef = new MTAFunction;
+tmpDef.label = "getCommandHandlers";
+tmpDef.description = "[ resource theResource ]";
+tmpDef.returnType = "table";
+tmpDef.args = ["[resource theResource]"];
+tmpDef.argDescs = {};
+tmpDef.scriptSide = ScriptSide.Client;
+ServerDefinitions.push(tmpDef);
+
+var tmpDef = new MTAFunction;
```
Comment (Subtixx) [inline src/defs/server.ts]: Why do you declare tmpDef here via var now?

File: src/defs/server.ts [hunk 2]
```diff
@@ -4,6 +4,51 @@ import {LuaFunction, MTAFunction, ScriptSide} from "./defs";
...
+tmpDef.label = "isResourceArchived";
+tmpDef.description = "Checks whether a resource is currently archived (running from within a ZIP file).";
+tmpDef.returnType = "bool";
+tmpDef.args = ["resource resourceElement"];
+tmpDef.argDescs = {};
+tmpDef.scriptSide = ScriptSide.Server;
+ServerDefinitions.push(tmpDef);
+
 var tmpDef = new MTAFunction;
```
Comment (Subtixx) [inline src/defs/server.ts]: And here? tmpDef should as the name says a temporary definition. I should've used let anyway, and I don't now why I didn't.

File: src/defs/server.ts [hunk 3]
```diff
@@ -904,24 +949,6 @@ tmpDef.argDescs = {};
 tmpDef.scriptSide = ScriptSide.Server;
 ServerDefinitions.push(tmpDef);
 
-tmpDef = new MTAFunction;
-tmpDef.label = "Vector2";
```
Comment (Subtixx) [inline src/defs/server.ts:908]: Why did you delete the vector and matrix constructors? They're a thing in MTA, just not properly implemented in vscode-mtalua: https://wiki.multitheftauto.com/wiki/Vector/Vector2

File: src/defs/shared.ts [hunk 4]
```diff
@@ -4,11 +4,110 @@ import {LuaFunction, MTAFunction, ScriptSide} from "./defs";
...
+tmpDef = new MTAFunction;
+tmpDef.label = "debugSleep";
+tmpDef.description = "WARNING: ONLY USE THIS FUNCTION IF YOU KNOW WHAT YOU ARE DOING! debugSleep freezes the client/server for the specified time. This means that all synchronization, rendering and script execution will stop except HTTP processing invoked by fetchRemote.";
+tmpDef.returnType = "bool";
+tmpDef.args = ["int miliseconds"];
+tmpDef.argDescs = {};
+tmpDef.scriptSide = ScriptSide.Client;
+SharedDefinitions.push(tmpDef);
+
+var tmpDef = new MTAFunction;
```
Comment (Subtixx) [inline src/defs/shared.ts]: Again why declare tmpDef in multiple spots?

File: src/defs/shared.ts [hunk 5]
```diff
@@ -4,11 +4,110 @@ import {LuaFunction, MTAFunction, ScriptSide} from "./defs";
...
+tmpDef.label = "getDevelopmentMode";
+tmpDef.description = "This function is used to get the development mode of the client. For more information see setDevelopmentMode.";
+tmpDef.returnType = "bool";
+tmpDef.args = [""];
+tmpDef.argDescs = {};
+tmpDef.scriptSide = ScriptSide.Shared;
+SharedDefinitions.push(tmpDef);
+
+var tmpDef = new MTAFunction;
```
Comment (Subtixx) [inline src/defs/shared.ts]: Declaring tmpDef again :)

File: src/defs/shared.ts [hunk 6]
```diff
@@ -4,11 +4,110 @@ import {LuaFunction, MTAFunction, ScriptSide} from "./defs";
...
+tmpDef.label = "setDevelopmentMode";
+tmpDef.description = "This function is used to set the development mode. Setting development mode allows access to special commands which can assist with script debugging.";
+tmpDef.returnType = "bool";
+tmpDef.args = ["bool enable", "[bool enableWeb = false]"];
+tmpDef.argDescs = {};
+tmpDef.scriptSide = ScriptSide.Shared;
+SharedDefinitions.push(tmpDef);
+
 var tmpDef = new MTAFunction;
```
Comment (Subtixx) [inline src/defs/shared.ts]: And again tmpDef declaration :)

File: src/defs/shared.ts [hunk 7]
```diff
@@ -1336,15 +1507,6 @@ tmpDef.argDescs = {};
 tmpDef.scriptSide = ScriptSide.Shared;
 SharedDefinitions.push(tmpDef);
 
-tmpDef = new MTAFunction;
-tmpDef.label = "getPedSkin";
```
Comment (Subtixx) [inline src/defs/shared.ts]: Shouldn't this say "DEPRECATED" instead of removing it completely? Also add a section for deprecated functions?

File: src/defs/shared.ts [hunk 8]
```diff
@@ -3433,15 +3550,6 @@ tmpDef.argDescs = {};
 tmpDef.scriptSide = ScriptSide.Shared;
 SharedDefinitions.push(tmpDef);
 
-tmpDef = new MTAFunction;
-tmpDef.label = "setPedFrozen";
```
Comment (Subtixx) [inline src/defs/shared.ts]: Again shouldn't this say "DEPRECATED" instead of removing it completely?

**Review by Subtixx — COMMENTED** [review]
First thanks for the pull request 👍 

Can you review the lines that I commented? Also in addition to adding them to Server.ts, Client.ts and Shared.ts those functions still have to be added to https://github.com/Subtixx/vscode-mtalua/blob/844f405205fb4aa00408ee476bd1346746e80282/syntaxes/mtalua.tmLanguage.json#L210-L221

File: src/defs/server.ts (hunk 1 above)
Comment (Loki214) [reply src/defs/server.ts]: most stuff I just copy pasted lol
sorry I know nothing about typeScript language

**Comment by Loki214** [comment]
Most stuff I just copy pasted lol
I'll undo the removal of the depreciated functions and add the new functions to the language.json
Sorry, I know nothing about the TypeScript language.

File: src/defs/server.ts (hunk 3 above)
Comment (Loki214) [reply src/defs/server.ts:908]: I didn't delete them I moved them to shared

**Comment by Loki214** [comment]
Done with all the changes

**Comment by Subtixx** [comment]
No worries. It wasn't mocking you, just some changes I would like to see before merging. But I think it now looks good.
